packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -102,12 +102,10 @@ const processNodeUrl = (
     originalUrl !== '/' &&
     htmlPath === '/index.html'
   ) {
-    const replacer = (url: string) =>
-      path.posix.join(
-        devBase,
-        path.posix.relative(originalUrl, devBase),
-        url.slice(1)
-      )
+    // #10990 config.base './' or '' will be changed to `/` in dev mode
+    // this function is only run in dev mode and config.base is always to be an absolute path.
+    // we needn't handle config.base './' or ''
+    const replacer = (url: string) => path.posix.join(devBase, url.slice(1))
```

Comment:
I don't think you need to strip the leading `/` as I believe `path.posix.join` will handle that for you
```suggestion
    const replacer = (url: string) => path.posix.join(devBase, url)
```

Reply:
That's right. Thank you!